osm2gtfs/core/elements.py

```diff
@@ -0,0 +1,187 @@
+        '''
+        Populates the object with information obtained from the tags
+        '''
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'colour' in self.tags:
+            self.route_color = self.tags['colour']
+
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'route_master' in self.tags:
+            self.route_type = self.tags['route_master'].capitalize()
+        else:
+            sys.stderr.write(
+                "Warning: Route master relation without a route_master tag:\n")
+            sys.stderr.write(" " + self.osm_url)
+
+            # Try to guess the type differently
+            if 'route' in self.tags:
+                self.route_type = self.tags['route'].capitalize()
+            else:
+                self.route_type = "Bus"
+
+    def add_itinerary(self, itinerary):
+
+        if self.route_id != itinerary.route_id:
+            raise ValueError('Itinerary route ID (' +
+                             itinerary.route_id +
+                             ') does not match Line route ID (' +
+                             self.route_id + ')')
+        # pylint: disable=no-member
+        self._itineraries.append(itinerary)
+
+    def get_itineraries(self):
+        return self._itineraries
+
+
+@attr.s
+class Itinerary(Element):
+    """A public transport service itinerary.
+
+    It's a representation of a possible variant of a line, grouped together by
+    a Line object.
+
+    In OpenStreetMap this is usually represented as "route" relation.
+    In GTFS this is not exlicitly presented but used as base to create "trips"
+
+    """
+    route_id = attr.ib()
+    shape = attr.ib()
+
+    line = attr.ib(default=None)
+    fr = attr.ib(default=None)
+    to = attr.ib(default=None)
+    duration = attr.ib(default=None)
+
+    # All stop objects of itinerary
+    stops = attr.ib(default=attr.Factory(list))
+
+    def __attrs_post_init__(self):
+        '''
+        Populates the object with information obtained from the tags
+        '''
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'from' in self.tags:
+            self.fr = self.tags['from']
+
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'to' in self.tags:
+            self.to = self.tags['to']
+
+    def get_stops(self):
+        return self.stops
+
+
+@attr.s
+class Station(Element):
+    """A public transport stop of the type station.
+
+    It's a representation of a possible group of stops.
+
+    In OpenStreetMap this is usually represented as "stop_area" relation.
+    In GTFS it is handled as a stop with a location_type=1. Regular Stops with
+    location_type=0 might specify a station as parent_station.
+
+    """
+    lat = attr.ib()
+    lon = attr.ib()
+
+    stop_id = attr.ib(default="")
+    location_type = attr.ib(default=1)
+    osm_url = attr.ib(default=None)
```

Comment:
`osm_url` is already inherited from `Element` class.

Reply:
Indeed. Fixed. Thanks for noticing.